src/microbe_stage/Microbe.Interior.cs

```diff
@@ -389,38 +389,34 @@ public Microbe Divide()
 
         var currentPosition = GlobalTransform.origin;
 
-        // Separate the two cells.
-        var separation = new Vector3(Radius, 0, 0);
+        // Find the direction to the right from where the cell is facing
+        var direction = GlobalTransform.basis.Quat().Normalized().Xform(Vector3.Right);
 
-        if (Colony != null)
-        {
-            // When in a colony we approximate a much higher separation distance
-            var colonyRadius = separation.x;
-
-            foreach (var colonyMember in Colony.ColonyMembers)
-            {
-                if (colonyMember == this)
-                    continue;
+        // Start calculating separation distance
+        var organellePositions = organelles!.Organelles.Select(o => Hex.AxialToCartesian(o.Position)).ToList();
 
-                var radius = colonyMember.Radius + Constants.COLONY_DIVIDE_EXTRA_DAUGHTER_OFFSET;
+        float distanceRight = MathUtils.GetMaximumDistanceInDirection(Vector3.Right, Vector3.Zero, organellePositions);
+        float distanceLeft = MathUtils.GetMaximumDistanceInDirection(Vector3.Left, Vector3.Zero, organellePositions);
 
-                // TODO: switch this to something else if this is too slow for large colonies
-                var positionInColony = colonyMember.GlobalTransform.origin - currentPosition;
+        if (Colony != null)
+        {
+            var colonyMembers = Colony.ColonyMembers.Select(c => c.GlobalTransform.origin).ToList();
 
-                var outerRadius = Math.Max(Math.Abs(positionInColony.x) + radius,
-                    Math.Abs(positionInColony.z) + radius);
+            distanceRight += MathUtils.GetMaximumDistanceInDirection(direction, currentPosition, colonyMembers);
```

Comment:
A slight improvement could be made here, GetMaximumDistanceInDirection could be modified to take in an IEnumerable instead of a list, that way `colonyMembers` won't need to be created into an actual list a few lines up from here. With that change GetMaximumDistanceInDirection can be called in some situations while saving the memory and GC pressure needed to create a temporary list.